lib/bookmarkable.rb

```diff
@@ -25,4 +25,12 @@ def update_bookmarker_pseuds_index
     return unless should_reindex_pseuds?
     IndexQueue.enqueue_ids(Pseud, bookmarks.pluck(:pseud_id), :background)
   end
+
+  def update_bookmarker_collections_index
+    return unless respond_to?(:should_reindex_collections?)
+    return unless should_reindex_collections?
+
+    collection_ids = bookmarks.map(&:collection_ids).flatten
```

Comment:
The potential runtime of this line worries me a little. This line of code generates N+1 database queries, and there are a few works with 10,000+ bookmarks, so N is potentially very large here. It might be good to reduce this to a single database query.

Reply:
I believe I replicated the logic with a single query. Does it look correct to you?